client/src/js/modules/dc/routes.js

```diff
@@ -99,6 +99,21 @@ let routes = [
         sgid: +route.params.sgid || null
     }),
   },
+  {
+    path: '/dc(/visit/)?:visit([a-zA-Z]{2}[0-9]+-[0-9]+)?(/dcg/)?:dcg([0-9]+)?(/page/)?:page([0-9]+)?(/ty/)?:ty([a-zA-Z0-9_-]+)?(/s/)?:search([a-zA-z0-9_-]+)?(/id/)?:id([0-9]+)?(/pjid/)?:pjid([0-9]+)?(/sgid/)?:sgid([0-9]+)?',
```

Comment:
I think we need to think about this, this could easily pro;iferate. Also typing "/" in the search really confuses it. I think we need to build the url and url encode the search term